src/models/UserModel.js

```diff
@@ -1,18 +1,80 @@
 import mongoose from 'mongoose';
 const Schema = mongoose.Schema;
 
-const UserSchema = new Schema({
-  email: {
-    type: String,
-    required: true,
-    unique: true
+const UserSchema = new Schema(
+  {
+    email: {
+      type: String,
+      required: true,
+      unique: true
+    },
+    password: {
+      type: String,
+      required: true,
+      unique: false
+    },
+    role: {
+      type: String,
+      required: true,
+      unique: false,
+      enum: ['admin', 'user'],
+      default: 'user'
+    },
+    firstName: {
+      type: String,
+      required: false,
+      unique: false
+    },
+    lastName: {
+      type: String,
+      required: false,
+      unique: false
+    },
+    // phone should be unique, but it will throw duplicate error if there's no value provided regardless null value
+    // solution: validate the uniqueness of phone in application logic or front-end
+    phone: {
+      type: String,
+      required: false
+    },
+    status: {
+      type: String,
+      required: false,
```

Comment:
The value of required should be true. Whenever someone creates a new account with us, we automatically set their account status to 'active'. So, right from the start, their account is ready to use. Does that make sense?

Reply:
or do you think we don't need to add that field in, since we have the default already?